### PR TITLE
feat: enable s3 sse default encryption

### LINKED ISSUE #17: RFE: enable encryption at rest where possible
All components of the stack (S3 SSE, Redis, RDS, ...) should use at least encryption at rest since users might be storing sensitive data depending on their legal requirements.

### PR DESCRIPTION
**Pull request by nodomain**
This MR enables default encryption with a KMS key for the S3 bucket as discussed in https://github.com/Rungutan/sentry-fargate-cf-stack/issues/17.

### CODE REVIEW

File: cloudformation-template.yaml [hunk 1]
```diff
@@ -1718,10 +1718,29 @@ Resources:
...
+              AWS:
+                - Fn::Sub: arn:aws:iam::${AWS::AccountId}:root
+
   SentryBucket:
     Type: AWS::S3::Bucket
     DeletionPolicy: Retain
     Properties:
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
```
Comment (mariusmitrofan) [inline cloudformation-template.yaml]: In order for the encryption to actually take place automatically, you have to add another property as per [AWS::S3::Bucket ServerSideEncryptionRule Documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-serversideencryptionrule.html)

Correct version is something along the lines of:

```
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - BucketKeyEnabled: true
            ServerSideEncryptionByDefault:
              KMSMasterKeyID:
                Fn::GetAtt: S3KmsKey.KeyId
              SSEAlgorithm: aws:kms
``` 

Aka the value of `BucketKeyEnabled` must be specifically set to **true** as it defaults to **false**.

**Review by mariusmitrofan — CHANGES_REQUESTED** [review]
See comment please about `BucketKeyEnabled` property.

Thank you!

File: cloudformation-template.yaml [hunk 2]
```diff
@@ -1718,10 +1718,29 @@ Resources:
       SecretString:
         Ref: SentryEmailPassword
 
+  S3KmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
```
Comment (mariusmitrofan) [inline cloudformation-template.yaml:1727]: The bucket policy will cause Sentry to error out due to it being unable to decrypt data from S3 once encrypted at rest.

A correct policy would look something along these lines:

```
  S3KmsKey:
    Type: AWS::KMS::Key
    Properties:
      EnableKeyRotation: true
      KeyPolicy:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Action: kms:*
            Resource: "*"
            Principal:
              AWS:
                - Fn::Sub: arn:aws:iam::${AWS::AccountId}:root
          - Effect: Allow
            Principal:
              AWS: "*"
            Action:
              - kms:Encrypt
              - kms:Decrypt
              - kms:ReEncrypt*
              - kms:GenerateDataKey*
              - kms:CreateGrant
              - kms:DescribeKey
            Resource: "*"
            Condition:
              StringEquals:
                kms:CallerAccount:
                  Ref: AWS::AccountId
                kms:ViaService:
                  Fn::Sub: s3.${AWS::Region}.amazonaws.com
```

Notice the 2nd statement to ensure that decryption can be possible.

Comment (nodomain) [reply cloudformation-template.yaml:1727]: Thanks for the hint!

File: cloudformation-template.yaml (hunk 1 above)
Comment (nodomain) [reply cloudformation-template.yaml]: Got you. Adding changes...

Comment (nodomain) [reply cloudformation-template.yaml]: Hopefully fixed with https://github.com/Rungutan/sentry-fargate-cf-stack/pull/18/commits/5444a2c86057c46ee01477bb2a9bfd234eac675e

**Review by mariusmitrofan — APPROVED** [review]

**Pull request merged**